client/components/email-verification/email-verification-gate.jsx

```diff
@@ -76,3 +49,14 @@ export default class EmailVerificationGate extends React.Component {
 		return <div>{ this.props.children }</div>;
 	}
 }
+
+export default connect(
+	( state ) => {
+		const user = getCurrentUser( state );
+		const site = getSelectedSite( state );
+		return {
+			userEmail: user && user.email,
+			needsVerification: userUtils.needsVerificationForSite( site )
```

Comment:
Does it make sense to use the `isCurrentUserEmailVerified` selector instead of `userUtils.needsVerificationForSite`?